admin/manufacturers.php

```diff
@@ -326,9 +326,12 @@
         </div>
         <!-- body_text_eof //-->
       </div>
-      <?php if (empty($action) && !empty($mInfo)) { ?>
+       <?php
+      if (empty($action)) {
+        $current_mid = (isset($mInfo)) ? 'mID=' . $mInfo->manufacturers_id : '';
+     ?>
         <div class="col-sm-12 text-right">
-          <a href="<?php echo zen_href_link(FILENAME_MANUFACTURERS, ($currentPage != 0 ? 'page=' . $currentPage . '&' : '') . 'mID=' . $mInfo->manufacturers_id . '&action=new'); ?>" class="btn btn-primary" role="button"><?php echo IMAGE_INSERT; ?></a>
+          <a href="<?php echo zen_href_link(FILENAME_MANUFACTURERS, ($currentPage != 0 ? 'page=' . $currentPage . '&' : '') . $current_mid . '&action=new'); ?>" class="btn btn-primary" role="button"><?php echo IMAGE_INSERT; ?></a>
```

Comment:
You'll need to move the `&` out of `'&action=new'`, into `$current_mid`, else you'll have double `&&` in some cases and a missing one in other cases.

Reply:
Thanks. So I can better understand why this is needed, and just how to implement it, I'll do some research for my own benefit and submit again once I have it figured out.

Reply:
I've updated to show better what I meant.

Here's how it works:
- note how the zen_href_link ends with checking and setting the page param with a trailing `&`
- now I've added to the end of your `$current_mid` assignment an `&` if/when it gets set
- and now the `action=new` param doesn't get a double-`&&` by mistake